labgrid/driver/bareboxdriver.py

```diff
@@ -108,6 +111,8 @@ def reset(self):
         self._status = 0
         self.console.sendline("reset")
         self._await_prompt()
+        if not self.boot_detected:
+            raise Exception(f"No reboot message detected: {self.boot_expression}")
```

Comment:
Please don't raise a plain Exception, this should be something along the lines of either ExecutionError or UBootStateError.